prisma/migrations/20250402214855_enable_rls/migration.sql

```diff
@@ -0,0 +1,274 @@
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all ApplicationSessions" ON "ApplicationSessions" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all LoginAttempts" ON "LoginAttempts" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Organizations" ON "Organizations" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Teams" ON "Teams" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Memberships" ON "Memberships" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+
+-- User-specific policies
+-- UserProfiles
+CREATE POLICY "Users can view their own profile" ON "UserProfiles"
+  FOR SELECT USING (auth.uid() = "supId");
+CREATE POLICY "Users can update their own profile" ON "UserProfiles"
+  FOR UPDATE USING (auth.uid() = "supId");
+
+-- Wallets
+CREATE POLICY "Users can view their own wallets" ON "Wallets"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can update their own wallets" ON "Wallets"
+  FOR UPDATE USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallets" ON "Wallets"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+CREATE POLICY "Users can delete their own wallets" ON "Wallets"
+  FOR DELETE USING (auth.uid() = "userId");
+
+-- WalletActivations
+CREATE POLICY "Users can view their own wallet activations" ON "WalletActivations"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallet activations" ON "WalletActivations"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+-- WalletRecoveries
+CREATE POLICY "Users can view their own wallet recoveries" ON "WalletRecoveries"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallet recoveries" ON "WalletRecoveries"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+-- WalletExports
+CREATE POLICY "Users can view their own wallet exports" ON "WalletExports"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallet exports" ON "WalletExports"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+-- WorkKeyShares
+CREATE POLICY "Users can view their own work key shares" ON "WorkKeyShares"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own work key shares" ON "WorkKeyShares"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+CREATE POLICY "Users can update their own work key shares" ON "WorkKeyShares"
+  FOR UPDATE USING (auth.uid() = "userId");
+
+-- RecoveryKeyShares
+CREATE POLICY "Users can view their own recovery key shares" ON "RecoveryKeyShares"
```

Comment:
For this one for example, deletes are allowed for the owner, because they might voluntarily get rid of an unused share or might be forced to discard one to create a new one.